Docs: convert boards to cmake

mbolivar opened the pull request:
Convert the ARM boards' usage of "make" commands to cmake using zephyr-app-commands.

With x86 done, this leaves ARC, NIOS II, RISC V 32, and XTensa. That's actually not very much work, since those arches don't have very many boards. That will come next, and boards/ will be converted.

Note that there are a few cases where "make" commands were deliberately left when building other tools.

Commit:
doc: boards: arm: convert make to cmake

Unify documentation formatting and use zephyr-app-commands where
applicable.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: boards: olimex: fix "resetting" typos

And add missing commas while we're here.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: boards: efm32wg_stk3800: fix note

The rst syntax is wrong; fix it. Fix a typo as well.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

carlescufi commented on boards/arm/frdm_k64f/doc/frdm_k64f.rst:
```diff
@@ -198,29 +198,30 @@ communication over USB.
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_k64f
+   :goals: build flash
```
should be :goals: flash if you want to replicate the original?

mbolivar replied:
Yes, I did these separately on purpose as I think it is good practice to  build first before running, but you're right that I shouldn't be doing that in this series. I'll revise all this.

carlescufi commented on boards/arm/frdm_kl25z/doc/frdm_kl25z.rst:
```diff
@@ -126,29 +126,29 @@ communication over USB.
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_kl25z
+   :goals: build flash
```
should be :goals: flash if you want to replicate the original?

carlescufi commented on boards/arm/hexiwear_k64/doc/hexiwear_k64.rst:
```diff
@@ -198,16 +203,12 @@ Flashing
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_k64
+   :goals: build flash
```
should be :goals: flash if you want to replicate the original?

carlescufi commented on boards/arm/qemu_cortex_m3/doc/board.rst:
```diff
@@ -71,9 +71,10 @@ Programming and Debugging
+.. zephyr-app-commands::
+   :zephyr-app: samples/synchronization
+   :board: qemu_cortex_m3
+   :goals: build run
```
run only

carlescufi reviewed (approved):
This is thorough and exhaustive, great work!. I have a few comments but they are minor.

mbolivar commented:
@carlescufi thanks for the review! Comments addressed.

mbolivar commented on boards/arm/stm32f3_disco/doc/stm32f3_disco.rst:
```diff
@@ -153,33 +153,24 @@ Flashing
-To build the Zephyr kernel and application, enter:
-
-.. code-block:: console
+First, connect the STM32F469I-DISCO Discovery kit to your host computer using
```
Wrong board, should be fixed

mbolivar commented on boards/arm/stm32f3_disco/doc/stm32f3_disco.rst:
```diff
@@ -95,7 +95,7 @@ Other hardware features are not yet supported on Zephyr porting.
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f3_disco/stm32f3_disco_defconfig``
+        ``boards/arm/stm32f3_disco/stm32f3_disco_defconfig``
```
Unrelated change

mbolivar commented on boards/arm/nucleo_l432kc/doc/nucleol432kc.rst:
```diff
@@ -168,50 +168,36 @@ Default settings are 115200 8N1.
+   $ minicom -D /dev/ttyACM0
 
-.. code-block:: console
+Now build and flash an applicaiton. Here is an example for
```
typo

mbolivar commented on boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst:
```diff
@@ -26,10 +26,10 @@ some highlights of the STM32F429I-DISC1 board:
-	- LD2 (red) for 3.3 V power-on
-	- Two user LEDs: LD3 (green), LD4 (red)
-	- Two USB OTG LEDs: LD5 (green) VBUS and LD6 (red) OC (over-current)
+        - LD1 (red/green) for USB communication
```
unrelated changes

mbolivar commented on boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst:
```diff
@@ -98,7 +98,7 @@ Other hardware features are not yet supported on Zephyr porting.
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig``
+        ``boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig``
```
unrelated

mbolivar commented on boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst:
```diff
@@ -136,6 +136,10 @@ The default communication settings are 115200 8N1.
 Programming and Debugging
 *************************
 
+Applications for the ``stm32f429i_disc1.rst`` board configuration can be built
```
delete .rst

mbolivar commented on boards/arm/stm32f469i_disco/doc/stm32f469i_disco.rst:
```diff
@@ -102,7 +102,7 @@ Other hardware features are not yet supported on Zephyr porting.
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig``
+        ``boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig``
```
unrelated

mbolivar commented on boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst:
```diff
@@ -189,62 +189,50 @@ Programming and Debugging
-Then follow instructions in README.md
+STM32L496G Discovery board includes an ST-LINK/V2-1 embedded debug
+tool interface.  This interface is supported by openocd version
+v0.10.0, which has been available since Zephyr SDK 0.9.2.
```
v0.9.2

mbolivar reviewed (commented):
Caught a few of my own issues, will push a fix momentarily

Commit:
doc: extensions: cosmetic whitespace fixes

Make flake8 happier about the whitespace.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: extensions: allow arbitrary goals

Rather than continuing to add build system goals, let's just trust the
user to do the right thing. The only special case is build, which is
the default goal.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: boards: arc: convert make to cmake

Use zephyr-app-commands appropriately.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: boards: nios2: convert make to cmake

Use zephyr-app-commands appropriately.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

Commit:
doc: boards: xtensa: convert make to cmake

Use zephyr-app-commands appropriately.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

mbolivar commented:
I had some time to get the rest of the boards today.

nashif commented on boards/arm/nucleo_f103rb/doc/nucleof103rb.rst:
```diff
@@ -124,33 +128,26 @@ This interface is supported by the openocd version included in the Zephyr SDK.
 
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic-blinky
+   :board: nucleof103rb
```
nucleo_f103rb?

nashif commented on boards/arm/nucleo_f103rb/doc/nucleof103rb.rst:
```diff
@@ -124,33 +128,26 @@ This interface is supported by the openocd version included in the Zephyr SDK.
-   $ make BOARD=nucleo_f103rb flash
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleof103rb
```
nucleo_f103rb

mbolivar replied:
Thanks for catching that. I was just copying the file name in as I assumed that was the convention after they stopped just being called "board.rst"

nashif commented on boards/arm/nucleo_f334r8/doc/nucleof334r8.rst:
```diff
@@ -128,45 +132,28 @@ This interface is supported by the openocd version included in Zephyr SDK.
-   $ minicom -D /dev/ttyACM0
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleof334r8
```
nucleo_f103rb, and many others below

mbolivar replied:
Seems you've copy pasted as well ;)

nashif reviewed (changes requested):
duh, wanted something else...

mbolivar commented:
@nashif nice catch! fixed.
